bolt/include/bolt/Core/DebugData.h

```diff
@@ -232,6 +238,8 @@ class DebugRangesSectionWriter {
   static constexpr uint64_t EmptyRangesOffset{0};
 
 private:
+  std::optional<DIE *> Die;
```

Comment:
Since we treat `nullptr` `Die` in the same was as unset `Die`, you can just have `DIE *Die{0};`.

Also add a comment describing what kind of DIE it is pointing to if any.